### PR TITLE
Enhancement/2585 adsense mock data

### PR DESCRIPTION
**Pull request by eugene-manuilov**
## Summary

Addresses issue #2585

## Checklist

- [ ] My code is tested and passes existing unit tests.
- [ ] My code has an appropriate set of unit tests which all pass.
- [x] My code is backward-compatible with WordPress 4.7 and PHP 5.6.
- [x] My code follows the [WordPress](https://make.wordpress.org/core/handbook/best-practices/coding-standards/) coding standards.
- [x] My code has proper inline documentation.
- [ ] I have added a QA Brief on the issue linked above.
- [x] I have signed the Contributor License Agreement (see <https://cla.developers.google.com/>).

### CODE REVIEW

File: assets/js/modules/adsense/util/data-mock.js
Comment (aaemnnosttv) [inline assets/js/modules/adsense/util/data-mock.js:176]: I would prefer to use `castArray` over `Array.isArray( value ) ? value : [ value ]` which we use in a few places already. I've seen this in a few places and I think it would be a bit easier to read then a longer line with inline ternary.

File: assets/js/modules/adsense/util/data-mock.js
Comment (aaemnnosttv) [inline assets/js/modules/adsense/util/data-mock.js:179]: I believe this is changing with v2 (not much I don't think) so we should probably hold on merging this until #3548 is merged since that one is more important.

File: assets/js/modules/adsense/util/data-mock.js
Comment (aaemnnosttv) [inline assets/js/modules/adsense/util/data-mock.js:197]: It looks like this is creating a new date incremented by `item` days from the start. How about something like this instead?

```js
new Date( startDate ).setDate( startDate.getDate() + item );
```

Also for formatting, if this is our date string, we should use our `getDateString` utility instead.

File: assets/js/modules/adsense/util/data-mock.js
Comment (aaemnnosttv) [inline assets/js/modules/adsense/util/data-mock.js:221]: Can we also add a `provideAdSenseMockReport` function here like we do for Analytics? That way we only need a single function to generate and receive the report which is what we'll need in most cases.

File: assets/js/modules/adsense/util/data-mock.js
Comment (aaemnnosttv) [inline assets/js/modules/adsense/util/data-mock.js:43]:
```suggestion
 * @param {Array.<string>} metrics    Metrics list.
```

File: assets/js/modules/adsense/util/data-mock.js
Comment (aaemnnosttv) [inline assets/js/modules/adsense/util/data-mock.js:199]:
```suggestion
		map( ( date ) => [
			...generateDimensionValues( date, dimensions ),
			...generateMetricValues( date, metrics ),
		] ),
```

**Review by aaemnnosttv — CHANGES_REQUESTED** [review]
Thanks @eugene-manuilov – this looks great overall. I left just a few comments mostly with suggestions, but I think we should probably wait to merge this until #3517 is merged to not delay or complicate the work in progress there.

Nice work!

File: assets/js/modules/adsense/util/data-mock.js
Comment (eugene-manuilov) [reply assets/js/modules/adsense/util/data-mock.js]: Nice, updated.

File: assets/js/modules/adsense/util/data-mock.js
Comment (eugene-manuilov) [reply assets/js/modules/adsense/util/data-mock.js]: Updated. :+1:

File: assets/js/modules/adsense/util/data-mock.js
Comment (eugene-manuilov) [reply assets/js/modules/adsense/util/data-mock.js:247]: Added.

File: assets/js/modules/adsense/util/data-mock.js
Comment (eugene-manuilov) [reply assets/js/modules/adsense/util/data-mock.js]: Updated.

File: assets/js/modules/adsense/util/data-mock.js
Comment (eugene-manuilov) [reply assets/js/modules/adsense/util/data-mock.js]: Updated.

File: assets/js/modules/adsense/util/data-mock.js
Comment (aaemnnosttv) [inline assets/js/modules/adsense/util/data-mock.js:259]: The Analytics provider (and other `provide*` utilities) don't mark resolvers as finished, so we should probably leave this out for consistency. It shouldn't have any effect to not do this since the report is only fetched if no report exists yet for the given options.
```suggestion
```

**Review by aaemnnosttv — CHANGES_REQUESTED** [review]
Looks good to me overall. I left one comment about `provideAdSenseMockReport` but nothing big. The main thing blocking this is the merge conflict which is a result of #3548 which is merged now. It looks like the only change with the new API for reports is the `kind` property has been removed. See https://github.com/google/site-kit-wp/pull/3548/files#diff-f99516d88faa1f82031cfc7876d0c93afeab899dcbda2572fc8ac9d24ccb3230

File: assets/js/modules/adsense/util/data-mock.js
Comment (eugene-manuilov) [reply assets/js/modules/adsense/util/data-mock.js]: Updated.

File: package-lock.json
Comment (aaemnnosttv) [inline package-lock.json:42959]: Can you see why this changed? It's minor but this is something I've noticed goes back and fourth in lock file updates. Dependabot PRs change this back to `https` as well. I'll fix it manually here to not block the PR but it's a bit odd.
```suggestion
          "resolved": "https://registry.npmjs.org/strip-ansi/-/strip-ansi-3.0.1.tgz",
```

File: package-lock.json
Comment (aaemnnosttv) [inline package-lock.json:36585]:
```suggestion
          "resolved": "https://registry.npmjs.org/strip-ansi/-/strip-ansi-3.0.1.tgz",
```

**Review by aaemnnosttv — APPROVED** [review]
Awesome!

**Pull request merged**
